core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/PartialHealthCheckStrategy.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint.healthcheck;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+import org.jctools.maps.NonBlockingHashSet;
+
+import com.linecorp.armeria.client.Endpoint;
+
+import com.google.common.collect.ImmutableSet;
+
+public class PartialHealthCheckStrategy implements HealthCheckStrategy {
+
+    private final Set<Endpoint> selectedCandidates;
+
+    private final double percentage;
+    private final int minimumSelectedCandidates;
+
+    private List<Endpoint> candidates;
+
+    private volatile boolean change;
+
+    public PartialHealthCheckStrategy(double percentage, int minimumSelectedCandidates) {
+        checkArgument(percentage > 0 && percentage <= 100, "percentage: %s (expected: 0.x-100)", percentage);
+        this.percentage = percentage;
```

Comment:
As we talked about, I made a builder.

But I removed minimum(number, percentage). 
Because it only works when use maximum percentage.

### ex
```
case1. set minium percentage 20%, maximum percentage 80%
when candidates are 5, selected candidates are 4. 
(The minimum percentage is not effective.)

case2. set minimum percentage 50%, maximum number 10;
when candidates are 6, seleced candidates is 6. (not 3)
(The minimum percentage is not effective.)

case3. set minimum number 5, maximum number 10
when candidates are 3, selected candidates are 3. (not 5)
when candidates are 8, selected candidates are 8. (not 5)
```

And I changed the percentage to ratio(0.x - 1).